### DISCUSSION TITLE
답장으로 챗을 가리거나 유저를 강퇴하는 방법

### DISCUSSION THREAD

**Discussion by hosot** [storycraft/node-kakao#781] [Q&A] [answered]
4버전에서 답장으로 답장한 메세지를 가리거나
답장으로 답장한 메시지의 유저를 강퇴하는 방법이 있을까요?

**Comment by gogop45** [comment, marked as answer]
답장형태로 가리기가능해요

**Reply by wiresharkkkk** [reply]
channel.getUserInfo({"userId":data.attachment().src_userId});
이렇게 해보세용

**Comment by gogop45** [comment]
![Uploading 20211104_114655.jpg…]()
![20211104_114655.jpg](https://user-images.githubusercontent.com/90106556/140249478-7e996eb9-ca0c-4916-b73d-d5e8e20c5dc0.jpg)

**Comment by hosot** [comment]
// 가리기

const reply = data.attachment<ReplyAttachment>();
    if (reply.src_userId !== undefined) {
      const srcUser =  channel.getUserInfo({"userId":reply.src_userId});
      console.log(srcUser);
    }
    if (reply.src_logId !== undefined) {
      if (reply.src_type !== undefined) {
        channel.hideChat({ 'logId': reply.src_logId, 'type': reply.src_type }).then().catch();   
        (async () => { await delay(200); })();
        channel.hideChat(data.chat).then().catch();   
      }
    }



// 추방
      const reply = data.attachment<ReplyAttachment>();
      /* 답장한 메시지 logId */
      const logId = reply.src_logId;
      /* 답장한 메시지의 원래 주인 ID */
      const reply_userId = reply.src_userId;

      if (reply.src_userId !== undefined) {
        const srcUser =  channel.getUserInfo({"userId":reply.src_userId});

        if (srcUser !== undefined) {

          channel.kickUser(srcUser).then().catch();   
          (async () => { await delay(200); })();
        }
      }

해결하였습니다 :)

**Comment by gogop45** [comment]
@hosot 님혹시 해외유저컷이나 감지해주는소스코드있나요?

**Reply by hosot** [reply]
저도 다른분들껄로 해결하긴했는데
이게 시간이 지나니까 결국 로그인하라고 또 나오더라구요..ㅠㅠ

**Reply by gogop45** [reply]
소스코드를 그대로 넣으면 되는건가요?
아니면 좀더 수정해야되는겁니까?
